jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp

```diff
@@ -458,25 +459,118 @@ ecma_builtin_date_now (ecma_value_t this_arg __attr_unused___) /**< this argumen
+  ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+  ecma_number_t *prim_value_num_p = NULL;
+
+  ecma_object_t *prototype_obj_p = ecma_builtin_get (ECMA_BUILTIN_ID_DATE_PROTOTYPE);
+  ecma_object_t *obj_p = ecma_create_object (prototype_obj_p,
+                                             true,
+                                             ECMA_OBJECT_TYPE_GENERAL);
+  ecma_deref_object (prototype_obj_p);
+
+  if (arguments_list_len == 0)
+  {
+    ECMA_TRY_CATCH (parse_res_value,
+                    ecma_builtin_date_now (ecma_make_object_value (obj_p)),
+                    ret_value);
+
+    prim_value_num_p = ecma_alloc_number ();
+    *prim_value_num_p = *ecma_get_number_from_value (parse_res_value);
+
+    ECMA_FINALIZE (parse_res_value)
+  }
+  else if (arguments_list_len == 1)
+  {
+    ECMA_TRY_CATCH (prim_comp_value,
+                    ecma_op_to_primitive (arguments_list_p[0], ECMA_PREFERRED_TYPE_NUMBER),
+                    ret_value);
+
+    if (ecma_is_value_string (prim_comp_value))
+    {
+      ECMA_TRY_CATCH (parse_res_value,
+                      ecma_builtin_date_parse (ecma_make_object_value (obj_p), prim_comp_value),
+                      ret_value);
+
+      prim_value_num_p = ecma_alloc_number ();
+      *prim_value_num_p = *ecma_get_number_from_value (parse_res_value);
+
+      ECMA_FINALIZE (parse_res_value);
+    }
+    else
+    {
+      ECMA_TRY_CATCH (prim_value, ecma_op_to_number (arguments_list_p[0]), ret_value);
+
+      prim_value_num_p = ecma_alloc_number ();
+      *prim_value_num_p = *ecma_get_number_from_value (prim_value);
+
+      ECMA_FINALIZE (prim_value);
+    }
+
+    ECMA_FINALIZE (prim_comp_value);
+  }
+  else if (arguments_list_len >= 2)
+  {
+    ECMA_TRY_CATCH (time_value,
+                    ecma_date_construct_helper (arguments_list_p, arguments_list_len),
+                    ret_value);
+
+    ecma_number_t *time_p = ecma_get_number_from_value (time_value);
+    prim_value_num_p = ecma_alloc_number ();
+    *prim_value_num_p = ecma_date_time_clip (ecma_date_utc (*time_p));
+
+    ECMA_FINALIZE (time_value);
+  }
+  else
+  {
+    prim_value_num_p = ecma_alloc_number ();
+    *prim_value_num_p = ECMA_NUMBER_ZERO;
+  }
+
+  if (ecma_is_completion_value_empty (ret_value))
+  {
+    ecma_property_t *class_prop_p = ecma_create_internal_property (obj_p,
+                                                                   ECMA_INTERNAL_PROPERTY_CLASS);
+    class_prop_p->u.internal_property.value = LIT_MAGIC_STRING_DATE_UL;
+
+    ecma_property_t *prim_value_prop_p = ecma_create_internal_property (obj_p,
+                                                                        ECMA_INTERNAL_PROPERTY_PRIMITIVE_NUMBER_VALUE);
+    ECMA_SET_POINTER (prim_value_prop_p->u.internal_property.value, prim_value_num_p);
+
+    ret_value = ecma_make_normal_completion_value (ecma_make_object_value (obj_p));
+  }
+  else if (ecma_is_completion_value_throw (ret_value))
```

Comment:
@ruben-ayrapetyan, sorry, I don't get it. What do you mean exactly? What would you like to test with the assert?